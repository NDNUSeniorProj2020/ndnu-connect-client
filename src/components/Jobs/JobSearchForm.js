import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Form, Button, Input, Typography } from 'antd';

const { Title } = Typography;
const { Search } = Input;

export default function JobSearchForm({ searchJobs, resetJobs }) {
	const [title, setTitle] = useState('');

	const onSubmit = (e) => {
		e.preventDefault();
		searchJobs(title);
	};

	const resetForm = () => {
		setTitle('');
		resetJobs();
	};

	return (
		<div>
			<Form id="job-search-form" onSubmit={onSubmit} layout="inline" style={{ textAlign: 'center' }}>
				<Title level={2} className="search-title">
					What job are you looking for?
          		</Title>
				<Search
					width={500}
					placeholder="What job are you looking for?"
					enterButton="Search"
					size="large"
					className="search-box"
					onSearch={value => console.log(value)}
				/>
				{/* <Form.Item>
					<Button id="submit-search-button" type="primary" onClick={onSubmit}>Search</Button>
				</Form.Item>
				<Form.Item>
					<Button id="reset-search-form-button" onClick={resetForm}>Reset</Button>
				</Form.Item> */}
			</Form>
		</div>
	);
}

JobSearchForm.propTypes = { searchJobs: PropTypes.func, resetJobs: PropTypes.func };
JobSearchForm.defaultProps = { searchJobs: f => f, resetJobs: f => f };
