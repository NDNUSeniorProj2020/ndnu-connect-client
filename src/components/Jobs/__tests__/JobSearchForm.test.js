import React from 'react';
import { shallow } from 'enzyme';

import JobSearchForm from '../JobSearchForm';

describe('tests for JobSearchForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<JobSearchForm />);
  });

	describe('snapshot tests', () => {
		it('renders without crashing', () => expect(wrapper).toMatchSnapshot());
	});

	describe('unit tests', () => {
		it('has default props', () => expect(JobSearchForm.defaultProps.searchJobs('f')).toEqual('f'));
  });

  describe('integration tests', () => {
    it('changes title state', () => {
      wrapper.find('#title-search-input').props().onChange({ target: { value: 'Receptionist' } });
    });

    it('submits form', () => {
      const e = { preventDefault: () => {} };
      wrapper.find("#job-search-form").props().onSubmit(e);
      wrapper.find("#submit-search-button").props().onClick(e);
    });

    it('resets form', () => wrapper.find('#reset-search-form-button').props().onClick());
  });
});
