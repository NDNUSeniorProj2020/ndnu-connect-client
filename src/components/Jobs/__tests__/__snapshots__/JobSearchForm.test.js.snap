// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for JobSearchForm snapshot tests renders without crashing 1`] = `ShallowWrapper {}`;
